lib/common/common/src/iterator_ext.rs

```diff
@@ -0,0 +1,69 @@
+pub trait IteratorExt: Iterator {
```

Comment:
Very neat abstraction, gonna use it as soon as it is merged :clap: 

Nitpick: I would love to have a unit test for it for a quick sanity check